convert2rhel/backup/packages.py

```diff
@@ -73,6 +74,145 @@
 )
 _UBI_9_REPO_PATH = os.path.join(_RHSM_TMP_DIR, "ubi_9.repo")
 
+# NOTE: Over time we want to replace this with pkghandler.RestorablePackageSet.
+class RestorablePackage(RestorableChange):
+    def __init__(self, pkgs, reposdir=None, set_releasever=False, custom_releasever=None, varsdir=None):
+        """Keep control of installed/removed RPM pkgs for backup/restore.
+
+        :param pkgs list[str]: List of packages to backup.
+        :param reposdir str: If a custom repository directory location needs to
+            be used, this parameter can be set with the location for it.
+        :param set_releasever bool: If there is need to set the relesever while
+            downloading the package with yumdownloader.
+        :param custom_releasever str: Custom releasever in case it need to be
+            overwritten and it differs from the `py:system_info.releasever`.
+        :param varsdir str: Location to the variables directory in case the
+            repository files needs to interpolate variables from those folders.
+        """
+        super(RestorablePackage, self).__init__()
+
+        self.pkgs = pkgs
+        self.reposdir = reposdir
+        self.set_releasever = set_releasever
+        self.custom_releasever = custom_releasever
+        self.varsdir = varsdir
+
+        self._backedup_pkgs_paths = []
+
+    def enable(self):
+        """Save version of RPMs packages."""
+        # Prevent multiple backup
+        if self.enabled:
+            return
+
+        loggerinst.info("Backing up the packages: %s." % ",".join(self.pkgs))
+        if os.path.isdir(BACKUP_DIR):
+            # If we detect that the current system is an EUS release, then we
+            # proceed to use the hardcoded_repofiles, otherwise, we use the
+            # custom reposdir that comes from the method parameter. This is
+            # mainly because of CentOS Linux which we have hardcoded repofiles.
+            # If we ever put Oracle Linux repofiles to ship with convert2rhel,
+            # them the second part of this condition can be dropped.
+            if system_info.eus_system and system_info.id == "centos":
+                self.reposdir = get_hardcoded_repofiles_dir()
+
+            # One of the reasons we hardcode repofiles pointing to archived
+            # repositories of older system minor versions is that we need to be
+            # able to download an older package version as a backup. Because for
+            # example the default repofiles on CentOS Linux 8.4 point only to
+            # 8.latest repositories that already don't contain 8.4 packages.
```

Comment:
I would move this comment a bit upper, above the `if system_info.eus_system and system_info.id == "centos":` condition. I think it's related to that one.

Reply:
Moved the comments to the docstring, think it keep more organized.